### ISSUE TITLE
native library naming.

### ISSUE THREAD

**Issue by harideva16** [thefangbear/JNI-By-Examples#12]
Java native interface (jni) allows calling native libraries

 1. Windows --> .dll,
 2. Linux --> .so and
 3. Mac -->.dylib.

I use the following code to load native library.

      static {
        System.loadLibrary("mypl");
        }

If I generate native library in windows os the native library's name comes `mypl.dll`

But , in Linux os the native library's name comes `libmypl.so` and in mac it comes `libmypl.dylib` 

Will this code ` System.loadLibrary("mypl");` load native library correcty after creating JPackage  in linux and mac ?

**Comment by denatarizk** [comment]
According to [this](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/design.html), it should.

**Comment by Krishnacoder78** [comment]
Yes, the code you've provided should work correctly after creating a JPackage distribution for Linux and macOS. The System.loadLibrary("mypl"); call is used to load a native library named "mypl". The naming convention you've mentioned for different operating systems is accurate:

Windows: mypl.dll
Linux: libmypl.so
macOS: libmypl.dylib
When you create a JPackage distribution for your Java application, it should include the necessary native libraries for each target operating system in the correct format. This means that the native library files should be named as per the conventions above and placed in the appropriate directories within the JPackage distribution package.

As long as the native library files are properly included and named in the distribution, the System.loadLibrary("mypl"); call should be able to locate and load the correct native library for the specific operating system on which your Java application is running, whether it's on Linux or macOS.

Just make sure that you structure your JPackage distribution correctly with the appropriate native library files, and the provided code snippet should work as expected.
